packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -238,3 +243,49 @@ export class ConflictDetector {
     telemetryService.sendException('ConflictDetectionException', errorMsg);
   }
 }
+
+export async function diffOneFile(
+  localFile: string,
+  remoteComponent: SourceComponent
+): Promise<void> {
+  const filePart = path.basename(localFile);
+
+  for (const filePath of remoteComponent.walkContent()) {
+    if (filePath.endsWith(filePart)) {
+      const remoteUri = vscode.Uri.file(filePath);
+      const localUri = vscode.Uri.file(localFile);
+
+      try {
+        await vscode.commands.executeCommand(
+          'vscode.diff',
+          remoteUri,
+          localUri,
+          'Source File Diff ( Local File / Org File )'
+        );
+      } catch (err) {
+        console.log(`ERROR: ${err}`);
```

Comment:
Do we want to show the error notification here rather than just logging it?

Reply:
this one should be taken care of in the single file pr https://github.com/forcedotcom/salesforcedx-vscode/pull/3240